exchanges/subscription/store.go

```diff
@@ -0,0 +1,187 @@
+package subscription
+
+import (
+	"fmt"
+	"maps"
+	"sync"
+
+	"github.com/thrasher-corp/gocryptotrader/common"
+)
+
+// Store is a container of subscription pointers
+type Store struct {
+	m  map[any]*Subscription
+	mu sync.RWMutex
+}
+
+// NewStore creates a ready to use store and should always be used
+func NewStore() *Store {
+	return &Store{
+		m: map[any]*Subscription{},
+	}
+}
+
+// NewStoreFromList creates a Store from a List
+func NewStoreFromList(l List) (*Store, error) {
+	s := NewStore()
+	for _, sub := range l {
+		if sub == nil {
+			return nil, common.ErrNilPointer
```

Comment:
Add context for all common.ErrNilPointer across this PR please.

Reply:
Done.

Various commits at the top of this.
Used riggrep on the diff to make sure all new ErrNilPointers have context.

This also led me to clarify thinking on a few of the use-cases.
I'll document that in a new PR that adds guidelines.
Essentially, though:
* Lazy init should be okay where possible
* nil catching should prevent panics
* TBC: We shouldn't be overzealous about erroring on nils, where we can ignore them
* TBC: We shouldn't have to sanity check state inside types, where New() methods exist and should be used

The last two merit confirmation so I'll review the last discussion with thrasher, put it into docs and PR them.